include/openssl/rsa.h

```diff
@@ -31,6 +31,7 @@ extern "C" {
 #  define OPENSSL_RSA_MAX_MODULUS_BITS   16384
 # endif
 
+# define OPENSSL_RSA_MIN_MODULUS_BITS	512
```

Comment:
On a *side* note related to *previous* #define. One can wonder if it's at all appropriate to have #ifndef OPENSSL_COMPILE_TIME_LIMIT in public header. I mean if intention is to inform application about libcrypto's compile-time limit, then one shouldn't "welcome" application to override it at its pre-processing time. It would be more honest to rather trigger warning.

Even though this is a *side* note, i.e. something one doesn't expect any action upon, one can project it to suggested modification by asking this. What is exact purpose of placing this macro to public header? Is it actually to inform application about the limit(*) or it it just single point of modification of essentially private compile-time limit? Because if it's actually latter, then rsa_locl.h might be more appropriate place for this.

(*) In other words would you expect application code to say `if (value < OPENSSL_RSA_MIN_MODULUS_BITS) { inform_user() }`? Keep in mind that application has to pay attention to return value [and inform user] in either case...

Reply:
> rsa_locl.h might be more appropriate place for this

Just in case, one can ask same question about other similar macros. Yes, of course. And one can even draw same conclusion about them too, i.e. that they don't belong in public header. Unfortunately we are not as free removing things from public header in non-major releases, as adding. In other words there is asymmetry, but one shouldn't use it as justification for doing something one regrets.

Reply:
Even though it's started as a side note, after some consideration, I [for one] can't find the decision to place the new macro to public rsa.h justified and effectively suggest to reconsider.

Reply:
a private header file should be ok as well.